### PR TITLE
Use file lock to avoid concurrent edits to dataset_description.json

### PR DESCRIPTION
**Pull request by cookpa**
## Changes proposed in this pull request

Use a file lock so that if multiple subjects are processed in parallel, the output `dataset_description.json` will only be written once.

Also fix the docker tag in the metadata, which referenced fmriprep.

I chose to make the lock file visible (ie, not starting with a .) because I figure it's best if it's visible just in case it causes problems and needs to be manually cleaned up. Happy to change this though

### CODE REVIEW

File: qsirecon/utils/bids.py [hunk 1]
```diff
@@ -185,7 +186,7 @@ def write_derivative_description(bids_dir, deriv_dir, dataset_links=None):
     if "QSIRECON_DOCKER_TAG" in os.environ:
         dset_desc["GeneratedBy"][0]["Container"] = {
             "Type": "docker",
-            "Tag": f"nipreps/fmriprep:{os.environ['QSIRECON_DOCKER_TAG']}",
+            "Tag": f"pennbbl/qsirecon:{os.environ['QSIRECON_DOCKER_TAG']}",
```
Comment (tsalo) [inline qsirecon/utils/bids.py:189]: I think qsirecon is being published to pennlinc instead of pennbbl

Comment (cookpa) [reply qsirecon/utils/bids.py:189]: Ah OK. @mattcieslak is it pennlinc? I will fix

Comment (cookpa) [reply qsirecon/utils/bids.py]: I forgot this is already built so I can check for myself on docker hub. Fixed

Comment (mattcieslak) [reply qsirecon/utils/bids.py]: Yes, we're moving these to pennlinc as part of The Great Split

**Review by mattcieslak — APPROVED** [review]
This looks great, thanks Phil!

**Comment by tsalo** [comment]
We should copy these changes over to QSIPrep, XCP-D, ASLPrep, etc.

**Pull request merged**